pkg/firewall/firewall_test.go

```diff
@@ -141,39 +126,31 @@ func TestConsultsAuthorizedOperatorsCache(t *testing.T) {
 	coreFirewall := newMockCoreFirewall()
 	policy := createNewPolicy(localChain, coreFirewall)
 
-	_, remotePeer1PublicKey, err := key.GenerateStaticNetworkKey()
-	if err != nil {
-		t.Fatal(err)
-	}
-	remotePeer1Address := common.HexToAddress(
-		key.NetworkPubKeyToChainAddress(remotePeer1PublicKey),
-	)
+	_, remotePeer1ID, remotePeer1Address := newPeer(t, localChain)
 
-	_, remotePeer2PublicKey, err := key.GenerateStaticNetworkKey()
-	if err != nil {
-		t.Fatal(err)
-	}
-	remotePeer2Address := common.HexToAddress(
-		key.NetworkPubKeyToChainAddress(remotePeer2PublicKey),
-	)
+	_, remotePeer2ID, remotePeer2Address := newPeer(t, localChain)
 
 	policy.authorizedOperatorsCache.Add(remotePeer1Address.String())
 
 	policy.nonAuthorizedOperatorsCache.Add(remotePeer2Address.String())
 	localChain.AuthorizeOperator(remotePeer2Address)
 
-	err = policy.validateAuthorization(localChain.PublicKeyToOperatorID(
-		(*ecdsa.PublicKey)(remotePeer1PublicKey),
-	))
-	if err != nil {
-		t.Errorf("expected no valdation error; has: [%v]", err)
+	expectedError := fmt.Sprintf(
+		"remote peer [%v] has no authorization on the factory",
+		remotePeer2ID,
+	)
+
+	if err := policy.validateAuthorization(remotePeer1ID); err != nil {
+		t.Errorf("expected no valdation error for remote peer 1; has: [%v]", err)
 	}
 
-	err = policy.validateAuthorization(localChain.PublicKeyToOperatorID(
-		(*ecdsa.PublicKey)(remotePeer2PublicKey),
-	))
-	if err != errNoAuthorization {
-		t.Errorf("expected error about no authorization; has: [%v]", err)
+	if err := policy.validateAuthorization(remotePeer2ID); err.Error() != expectedError {
+		t.Errorf(
+			"unexpected error about no authorization for remote peer 2\n"+
+				"actual:   [%v]\nexpected: [%v]",
```

Comment:
I like having `[]` in the log messages - it lets a human know when something is variable, and it makes it easier to write programs that parse the logs.

I'm not sure that I like having `[]` in the test error classes though. It feels like:
```
unexpected error about no authorization for remote peer 2
actual:   banana
expected: fofana
```
is easier to read than
```
unexpected error about no authorization for remote peer 2
actual:   [banana]
expected: [fofana]
```

That said, this isn't scoped to this PR, so I won't block on it!

Reply:
Also, in the liquidation recovery stuff, we always put `expected` before `actual`:

```go
			if err == nil {
				t.Errorf(
					"unexpected error message\nexpected: %s\nactual:   nil",
					testData.err,
				)
```